File: Client/Media/Captions/CaptionsLib/source/tts/TTSWriter.cs
//Filename: TTSWriter.cs
//Version: 20131105

using Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions;
using System.IO;

namespace ClipFlair.CaptionsLib.TTS
{

	public class TTSWriter : BaseCaptionWriter
	{

		#region --- Methods ---

		public override void WriteCaption(CaptionElement caption, TextWriter writer)
		{
			writer.WriteLine(TTSUtils.CaptionToTTSString(caption));
		}

		#endregion

	}

}

File: Client/Media/Captions/CaptionsLib/VB/Source/tts/TTSWriter.vb
'Filename: TTSWriter.vb
'Version: 20131105

Imports ClipFlair.CaptionsLib.TTS.TTSUtils

Imports System.IO
Imports Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions

Namespace ClipFlair.CaptionsLib.TTS

  Public Class TTSWriter
    Inherits BaseCaptionWriter

#Region "--- Methods ---"

    Public Overrides Sub WriteCaption(ByVal caption As CaptionElement, ByVal writer As TextWriter)
      writer.WriteLine(CaptionToTTSString(caption))
    End Sub

#End Region

  End Class

End Namespace
